rules/onchain_budget.go

```diff
@@ -0,0 +1,694 @@
+			"required amount")
+	}
+
+	if o.MaxSatPerVByte < minOCB.MaxSatPerVByte {
+		return fmt.Errorf("on-chain fee rate limit is below the " +
+			"minimum required amount")
+	}
+
+	return nil
+}
+
+// RuleName returns the name of the rule that these values are to be used with.
+//
+// NOTE: this is part of the Values interface.
+func (o *OnChainBudget) RuleName() string {
+	return OnChainBudgetName
+}
+
+// ToProto converts the rule Values to the litrpc counterpart.
+//
+// NOTE: this is part of the Values interface.
+func (o *OnChainBudget) ToProto() *litrpc.RuleValue {
+	return &litrpc.RuleValue{
+		Value: &litrpc.RuleValue_OnChainBudget{
+			OnChainBudget: &litrpc.OnChainBudget{
+				AbsoluteAmtSats: o.AbsoluteAmtSats,
+				MaxSatPerVByte:  o.MaxSatPerVByte,
+			},
+		},
+	}
+}
+
+// PseudoToReal attempts to convert any appropriate pseudo fields in the rule
+// Values to their corresponding real values. It uses the passed PrivacyMapDB to
+// find the real values. This is a no-op for the OnChainBudget rule.
+//
+// NOTE: this is part of the Values interface.
+func (o *OnChainBudget) PseudoToReal(_ firewalldb.PrivacyMapDB,
+	_ session.PrivacyFlags) (Values, error) {
+
+	return o, nil
+}
+
+// RealToPseudo converts the rule Values to a new one that uses pseudo keys,
+// channel IDs, channel points etc. It returns a map of real to pseudo strings
+// that should be persisted. This is a no-op for the OnChainBudget rule.
+//
+// NOTE: this is part of the Values interface.
+func (o *OnChainBudget) RealToPseudo(db firewalldb.PrivacyMapReader,
+	flags session.PrivacyFlags) (Values, map[string]string, error) {
+
+	return o, nil, nil
+}
+
+// checkFeeRate is a helper function that returns an error if the given fee rate
+// exceeds the maximum allowed by the rule that the enforcer is enforcing.
+func (o *OnChainBudgetEnforcer) checkFeeRate(satPerVByte uint64) error {
+	if satPerVByte > o.MaxSatPerVByte {
+		return fmt.Errorf("max fee rate exceeded")
+	}
+	return nil
+}
+
+// formatReqId creates a unique identifier for the request that can be used to
+// associate the action accross request-response handling.
+func formatReqId(lndConnID string, reqID int64) string {
+	return fmt.Sprintf("%s-%d", lndConnID, reqID)
+}
+
+// handleOpenChannelRequest is a helper function which handles the request for
+// opening a single channel.
+func (o *OnChainBudgetEnforcer) handleOpenChannelRequest(ctx context.Context,
+	r *lnrpc.OpenChannelRequest) (proto.Message, error) {
+
+	if r.FundMax {
+		return nil, fmt.Errorf("max funding not supported")
+	}
+
+	if len(r.Outpoints) > 0 {
+		return nil, fmt.Errorf("outpoints not supported")
+	}
+
+	if r.TargetConf != 0 {
+		return nil, fmt.Errorf("target conf not supported")
+	}
+
+	err := o.checkFeeRate(r.SatPerVbyte)
+	if err != nil {
+		return nil, err
+	}
+
+	// Derive a unique identifier for the request to be able to associate it
+	// with the repsonse.
+	reqID := formatReqId(o.GetLndConnID(), o.GetReqID())
```

Comment:
nit: I think it's a little confusing that we name this variable to `reqID` here and at other places, as that can be confused with the request ID returned by the `GetReqID` function. 
Perhaps it should be renamed to something like `prefReqID`?